lib/shipengine/domain/carriers.rb

```diff
@@ -1,64 +1,93 @@
 # frozen_string_literal: true
 
-require "shipengine/utils/validate"
+require "hashie"
+require_relative "carriers/list_carriers"
 require "shipengine/constants"
-require "shipengine/utils/pretty_print"
+require "pry"
 
 module ShipEngine
-  class Carrier
-    include ::ShipEngine::Utils::PrettyPrint
-    attr_reader :code, :name
-
-    # @param code [string] e.g. 'ups' | 'fedex'
-    def initialize(code)
-      @code = code
-      @name = CARRIER_MAP[code]
-    end
-
-    # {"carrier_code" => "name"}
-    CARRIER_MAP = {
-      "ups" => "United Parcel Service",
-      "fedex" => "FedEx",
-      "usps" => "U.S. Postal Service",
-      "stamps_com" => "Stamps.com",
-    }.freeze
-    # TODO: update above carrier map to be current, missing several carriers.
-  end
-
-  class CarrierAccount
-    include ::ShipEngine::Utils::PrettyPrint
-    attr_reader :carrier, :account_id, :account_number, :name
-
-    # @param carrier [Carrier]
-    # @param account_id [String] The unique ID that is associated with the current carrier account.
-    # @param account_number [String] The account number of the current carrier account.
-    # @param name [String] e.g. The account name of the current carrier account.
-    def initialize(carrier_code:, account_id:, account_number:, name:)
-      @carrier = Carrier.new(carrier_code)
-      @account_id = account_id
-      @account_number = account_number
-      @name = name
-    end
-  end
-
   module Domain
-    class Carrier
+    class Carriers
       # @param [ShipEngine::InternalClient] internal_client
       def initialize(internal_client)
         @internal_client = internal_client
       end
 
-      def list_accounts(config:, carrier_code: nil)
-        response = @internal_client.make_request("carrier.listAccounts.v1", { carrierCode: carrier_code }.compact, config)
-        accounts = response.result["carrierAccounts"]
-        accounts.map do |account|
-          CarrierAccount.new(
-            carrier_code: account["carrierCode"],
-            account_id: account["accountID"],
-            account_number: account["accountNumber"],
-            name: account["name"]
+      def list_carriers(config:)
+        response = @internal_client.get("/v1/carriers", {}, config)
+        carriers_api_result = response.body
+        mash_result = Hashie::Mash.new(carriers_api_result)
```

Comment:
- Maybe we just bite the bullet and have internal_client return a Hashie::Mash if there's a response body, since it's the body is always going to be serializable AFAIK?
- Is Hashie just so you can use the dot (.) operator to access arbitrary fields?